sensorlib/doc/CarlaCDASimAPI.md

```diff
@@ -0,0 +1,208 @@
+accessing features is through a CarlaCDASimAPI object, which may be constructed from any piece of information sufficient to
+establish a Carla connection.
+
+First ensure the following environment variables are set which define the CARLA version being utilized:
+
+| Environment Variable | Description                                                                                                       |
+| -------------------- |-------------------------------------------------------------------------------------------------------------------|
+| LOAD_CARLA_EGG       | Set to True to load the CARLA Python API from an egg file. Leave blank to use a pip-managed CARLA client library. |
+| CARLA_VERSION        | The CARLA version to load (ex: "0.0.14").                                                                         |
+| CARLA_EGG_DIR        | (optional) Hint directory for searching for the CARLA egg file.                                                   |
+
+
+Nex, build the API object.
+
+```
+api = CarlaCDASimAPI.build_from_host_spec(carla_host, carla_port)
+
+# or...
+
+api = CarlaCDASimAPI.build_from_client(carla_client)
+
+# or...
+
+api = CarlaCDASimAPI.build_from_world(carla_world)
+```
+
+The API object may then be used to indirectly build and manage sensors, and request output data. Sensors are located by
+association with an infrastructure item, such as a traffic light post, and the sensor index within a co-located
+collection of sensors. For example, a sensor with an infrastructure_id of 3 and sensor_id 0, is the 0th sensor
+associated with infrastructure item 3.
+
+The following sample loads configurations and constructs a sensor.
+
+```
+api = CarlaCDASimAPI.build_from_world(world)
+
+infrastructure_id = 3
+sensor_id = 7
+detection_cycle_delay_seconds = 0.5
+sensor_config  =  SimulatedSensorUtils.load_config_from_file(arg.sensor_config_filename)
+simulated_sensor_config = sensor_config["simulated_sensor"]
+carla_sensor_config = sensor_config["lidar_sensor"]
+noise_model_config = SimulatedSensorUtils.load_config_from_file(arg.noise_model_config_filename)
+user_offset = carla.Location(arg.x, arg.y, arg.z)
+lidar_transform = carla.Transform(carla.Location(x=-0.5, z=1.8) + user_offset)
+
+sensor = api.create_simulated_semantic_lidar_sensor(simulated_sensor_config, carla_sensor_config, noise_model_config,
+                                                    detection_cycle_delay_seconds,
+                                                    infrastructure_id, sensor_id,
+                                                    lidar_transform.location, lidar_transform.rotation, vehicle.id)
+
+```
+
+Objects can be retrieved from the sensor directly:
+
+```
+detected_objects = sensor.get_detected_objects()
+```
+
+Objects can also be retrieved from another function context by using the API. (Note: Currently sensor access from the
+API is only supported within the running Python context.)
```

Comment:
What distinction are you describing here.

Reply:
Context.